service/datasink-service.py

```diff
@@ -16,9 +17,11 @@ def is_2xx_status(response):
 @app.route('/receiver', methods=['POST'])
 def receiver():
     # get entities from request and write each of them to a file
-
     headers = {"content-type": "application/json"}
-    r = requests.post(API_ENDPOINT, data=request.data, headers=headers)
+    if USERNAME != None and PASSWORD != None:
```

Comment:
I think, sending `auth=(None, None)` is equivalent to not sending _auth_ at all. Hence, i doubt we need an if statement here. This is a coding style taste, though.

Reply:
oauth2session.py line:403 uses the auth parameter like this:
auth = kwargs.pop('auth', None)
where None is the value returned if 'auth' is not present and None != (None, None) so I don't think that would work. 
But yeah as you said if it did work then it would not change the program behaviour.